lib/internal/test_runner/test.js

```diff
@@ -548,7 +539,10 @@ class Test extends AsyncResource {
       const runArgs = ArrayPrototypeSlice(args);
       ArrayPrototypeUnshift(runArgs, this.fn, ctx);
 
-      if (this.fn.length === runArgs.length - 1) {
+      if (this.fn === noop && this.parent === null) {
```

Comment:
Do we need the `this.fn === noop` part of this check? If we're looking for the root test, checking the parent should be enough.

Reply:
`this.parent === null` is also true for hooks

Reply:
Ah ok. What about `this.root === this`?

Reply:
I removed this condition since the missing await solved that